data_model/src/expression.rs

```diff
@@ -338,962 +499,561 @@ impl ContextValue {
+    /// Works only for [`Value::U32`].
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{} > {}",
+        "self.left.parenthesise(Operation::Greater)",
+        "self.right.parenthesise(Operation::Greater)"
+    )]
+    pub Greater(u32, u32) -> bool
+}
+
+gen_expr_and_impls! {
+    /// Returns whether the `left` expression is less than the `right`.
+    /// Works only for [`Value::U32`].
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{} < {}",
+        "self.left.parenthesise(Operation::Less)",
+        "self.right.parenthesise(Operation::Less)"
+    )]
+    pub Less(u32, u32) -> bool
+}
+
+gen_expr_and_impls! {
+    /// Negates the result of the `expression`.
+    /// Works only for `Value::Bool`.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(fmt = "!{}", "self.expression.parenthesise(Operation::Not)")]
+    pub Not(bool) -> bool
+}
+
+gen_expr_and_impls! {
+    /// Applies the logical `and` to two `Value::Bool` operands.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{} && {}",
+        "self.left.parenthesise(Operation::And)",
+        "self.right.parenthesise(Operation::And)"
+    )]
+    pub And(bool, bool) -> bool
+}
+
+gen_expr_and_impls! {
+    /// Applies the logical `or` to two `Value::Bool` operands.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
+        fmt = "{} || {}",
+        "self.left.parenthesise(Operation::Or)",
+        "self.right.parenthesise(Operation::Or)"
+    )]
+    pub Or(bool, bool) -> bool
 }
 
-impl Multiply {
-    /// Number of underneath expressions.
-    pub fn len(&self) -> usize {
-        self.left.len() + self.right.len() + 1
-    }
+/// Builder for [`If`] expression.
+#[derive(Debug)]
+#[must_use = ".build() not used"]
+pub struct IfBuilder {
+    /// Condition expression, which should evaluate to `Value::Bool`.
+    /// If it is `true` then the evaluated `then_expression` will be returned, else - evaluated `else_expression`.
```

Comment:
```suggestion
    /// If it is `true`, then the evaluated `then_expression` is returned. Otherwise, the evaluated `else_expression` is returned.
```